assets/product-info.js

```diff
@@ -91,13 +93,34 @@ if (!customElements.get('product-info')) {
         this.#renderProductInfo(targetUrl, variant?.id, targetId, callback);
       }
 
+      // Load the product page and update sections
+      #updateProductSections(url, variantId) {
+        this.sectionRefreshAbortController?.abort();
+        this.sectionRefreshAbortController = new AbortController();
+        fetch(this.#getProductInfoUrl(url, variantId, true), {
+          signal: this.sectionRefreshAbortController.signal,
+        })
+          .then((response) => response.text())
+          .then((responseText) => {
+            const html = new DOMParser().parseFromString(responseText, 'text/html');
+            const updatedSections = Array.from(html.querySelectorAll('main > .shopify-section')).reduce(
+              (sectionsById, section) => ({ ...sectionsById, [section.id]: section }),
+              {}
+            );
+
+            document.querySelectorAll('main > .shopify-section').forEach((section) => {
+              if (!section.id.includes('__main') && updatedSections[section.id]) {
+                this.swapProductUtility.viewTransition(section, updatedSections[section.id]);
+              }
+            });
+          });
+      }
```

Comment:
This function is pretty naive and assumes that sibling products of a combined listing will use the same product template. If the product sections differ the UX will be pretty poor.

On the other hand, the primary use case for combined listings is to combine many products to appear as a single product. Because of this, the functionality between combined listing children should be identical in most cases, and children having different content should be a slim edge case.

Given that this isn't a one-way door (e.g. merchants can update theme code to accommodate multi-template CLs) and that it is an edge case, our designer on combined listings thought this was an acceptable tradeoff.

What do you think?